src/Service/LocaleProviderService.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupBundle\Service;
+
+interface LocaleProviderService
+{
+    /**
+     * Determine the current users preferred locale, may be empty.
+     *
+     * @return string
+     *   Locale like nl_NL or just an empty string if indeterminate.
```

Comment:
You're already potentially returning `null` in Gateway's SecondFactorLocaleProvider, why return an empty string and not `null`?